src/csm_testing/tests/sat_functional/hwinv/test_hwinv.py

```diff
@@ -0,0 +1,632 @@
+        stderr = proc.stderr.decode()
+
+        self.assertIn("Listing of all router bmcs", stdout)
+        actual_header = get_column_names_list(stdout)
+        expected_header = adjust_expected_header(HWINV_ROUTER_BMCS_HEADER, stderr)
+
+        if not actual_header:
+            return
+
+        self.assertEqual(expected_header, actual_header)
+
+    def test_hwinv_list_mgmt_switches(self) -> None:
+        """Test that `sat hwinv --list-mgmt-switches` returns proper header"""
+        command = 'sat hwinv --list-mgmt-switches'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        self.assertIn("Listing of all mgmt switches", stdout)
+        actual_header = get_column_names_list(stdout)
+        expected_header = adjust_expected_header(HWINV_MGMT_SWITCHES_HEADER, stderr)
+
+        if not actual_header:
+            return
+
+        self.assertEqual(expected_header, actual_header)
+
+    def test_hwinv_list_cabinet_pdus(self) -> None:
+        """Test that `sat hwinv --list-cabinet-pdus` returns proper header"""
+        command = 'sat hwinv --list-cabinet-pdus'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        self.assertIn("Listing of all cabinet pdus", stdout)
+        actual_header = get_column_names_list(stdout)
+        expected_header = adjust_expected_header(HWINV_CABINET_PDUS_HEADER, stderr)
+
+        if not actual_header:
+            return
+
+        self.assertEqual(expected_header, actual_header)
+
+    def test_hwinv_list_cabinet_pdu_power_connectors(self) -> None:
+        """Test that `sat hwinv --list-cabinet-pdu-power-connectors` returns proper header"""
+        command = 'sat hwinv --list-cabinet-pdu-power-connectors'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        self.assertIn("Listing of all cabinet pdu power connectors", stdout)
+        actual_header = get_column_names_list(stdout)
+        expected_header = adjust_expected_header(HWINV_CABINET_PDU_POWER_CONNECTORS_HEADER, stderr)
+
+        if not actual_header:
+            return
+
+        self.assertEqual(expected_header, actual_header)
+
+    def test_hwinv_list_all(self) -> None:
+        """Test that `sat hwinv --list-all` prints all expected sections and their headers."""
+        command = 'sat hwinv --list-all'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        section_headers = {
+            "Listing of all nodes": HWINV_NODE_HEADER,
```

Comment:
I would suggest removing the repeated "Listing of all" part of the keys and just use the human-friendly component type. E.g.:

```suggestion
            "nodes": HWINV_NODE_HEADER,
```